codecov_auth/signals.py

```diff
@@ -7,41 +9,86 @@
 
 from codecov_auth.models import OrganizationLevelToken, Owner, OwnerProfile
 
+log = logging.getLogger(__name__)
+
 
 @receiver(post_save, sender=Owner)
 def create_owner_profile_when_owner_is_created(
-    sender, instance: Owner, created, **kwargs
-):
+    sender: Type[Owner], instance: Owner, created: bool, **kwargs: Dict[str, Any]
+) -> Optional[OwnerProfile]:
     if created:
         return OwnerProfile.objects.create(owner_id=instance.ownerid)
 
 
-_pubsub_publisher = None
+class ShelterPubsub:
+    pubsub_publisher = None
+    _instance = None
+
+    @classmethod
+    def get_instance(cls) -> "ShelterPubsub":
+        """
+        This class needs the Django settings to be fully loaded before it can be instantiated,
+        therefore use this method to get an instance rather than instantiating directly.
+        """
+        if cls._instance is None:
+            cls._instance = cls()
+        return cls._instance
+
+    def __init__(self) -> None:
+        if not self.pubsub_publisher:
+            self.pubsub_publisher = pubsub_v1.PublisherClient()
+        pubsub_project_id: str = settings.SHELTER_PUBSUB_PROJECT_ID
 
+        # topic_id has REPO in the name but it is used for all types of objects
+        topic_id: str = settings.SHELTER_PUBSUB_SYNC_REPO_TOPIC_ID
+        self.topic_path = self.pubsub_publisher.topic_path(pubsub_project_id, topic_id)
 
-def _get_pubsub_publisher():
-    global _pubsub_publisher
-    if not _pubsub_publisher:
-        _pubsub_publisher = pubsub_v1.PublisherClient()
-    return _pubsub_publisher
+    def publish(self, data: Dict[str, Any]) -> None:
```

Comment:
Nice! This looks great! A last nitpick, could we move this to somewhere like `utils/shelter.py` in the root folder, or anywhere you see fit? This isn't strictly a codecov_auth responsibility so it makes sense it is its own standalone thing

Reply:
agree! moved!